pkg/internal/storage/funder/result.go

```diff
@@ -0,0 +1,31 @@
+package funder
+
+import (
+	"fmt"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/satoshi"
+	"github.com/go-softwarelab/common/pkg/seq"
+)
+
+type Result struct {
+	AllocatedUTXOs []*UTXO
+	ChangeCount    uint64
+	ChangeAmount   satoshi.Value
+	Fee            satoshi.Value
+}
+
+func (fr *Result) TotalAllocated() (satoshi.Value, error) {
+	total, err := satoshi.Sum(seq.Map(seq.FromSlice(fr.AllocatedUTXOs), func(utxo *UTXO) satoshi.Value {
+		return utxo.Satoshis
+	}))
```

Comment:
```suggestion
        if fr.AllocatedUTXOs == nil {
		return 0, nil // Explicitly handle nil slice gracefully
	}
	total, err := satoshi.Sum(seq.Map(seq.FromSlice(fr.AllocatedUTXOs), func(utxo *UTXO) satoshi.Value {
		return utxo.Satoshis
	}))
```
if `fr.AllocatedUTXOs` is nil, `seq.FromSlice` will still work (returns empty slice), but it might be good to explicitly check and return early ? WDYT ?

Reply:
seq.FromSlice is not returning slice, 
It is iterator, so it's simply iterating over the fr.AllocatedUTXOs
and it's working the same as:
```go
for _, v := fr.AllocatedUTXOs {
```
and if it is nil, then for loop won't start
and because there is no need to explicitly check the len of the slice before for loop, I don't think that this is needed in this case.